### PR TITLE
Use new table search with filters

### PR DESCRIPTION
**Pull request by SferaDev**
``api.records.searchTable(workspace, database, branch, table, { query: "foo", filter: { name: "bar" } })``

``xata.db.table.search("foo", { filter: { name: "bar" } })``

### CODE REVIEW

File: packages/client/src/schema/repository.ts [hunk 1]
```diff
@@ -453,12 +458,19 @@ export class RestRepository<Data extends BaseData, Record extends XataRecord = D
...
 
-    const { records } = await searchBranch({
-      pathParams: { workspace: '{workspaceId}', dbBranchName: '{dbBranch}' },
-      body: { tables: [this.#table], query, fuzziness: options.fuzziness },
+    const { records } = await searchTable({
+      pathParams: { workspace: '{workspaceId}', dbBranchName: '{dbBranch}', tableName: this.#table },
+      body: {
+        query,
+        fuzziness: options.fuzziness,
+        filter: options.filter as Schemas.FilterExpression
```
Comment (SferaDev) [inline packages/client/src/schema/repository.ts:472]: I need to battle with these types. Opening https://github.com/xataio/client-ts/issues/313

File: cli/src/commands/shell/index.ts [hunk 2]
```diff
@@ -154,13 +154,17 @@ function getDefaultEval() {
...
+  options: { table: boolean; deep?: boolean },
+  callback: (err: Error | null, result: any) => void
+) {
   const { table } = options;
   return callback(
     null,
     table
       ? console.table(result)
-      : typeof result === 'object'
+      : typeof result === 'object' && options.deep
```
Comment (gimenete) [inline cli/src/commands/shell/index.ts:167]: Why are you adding the `deep` flag? Don't we want that not only in raw API calls, but in JS statements as well?

**Review by gimenete — APPROVED** [review]
LGTM. Just one question.

File: cli/src/commands/shell/index.ts (hunk 2 above)
Comment (SferaDev) [reply cli/src/commands/shell/index.ts:167]: Tudor noticed that autocomplete is also using postprocessing and had some weird behavior. I'll forward you the video.

Added it as an opt-in flag, and let's decide where to add it.

**Pull request merged**